Grids AI Assistant: Add How-To Topics (#8819)
DevExpress/devextreme-documentation: concepts/05 UI Components/DataGrid/12 AI Assistant/15 Specifics and Best Practices.md

Note the following AI Assistant specifics and best practices:

> - The assistant does not have access to component data out of the box. In certain usage scenarios, this approach can cause commands to fail.

Reviewer: ```suggestion
- The assistant does not have access to component data by default. In some scenarios, this can cause commands to fail.
```

>     For instance, to select the last row on a page (when [paging](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/paging/) is enabled), the assistant calls the `selectByIndexes` command and uses the page size to specify an index. If the number of rows on the active page is smaller than the page size, the command fails.

Reviewer 2: ```suggestion
    For instance, to select the last row on a [page](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/paging/), the assistant calls the `selectByIndexes` command and uses the page size to specify an index. If the number of rows on the active page is smaller than the page size, the command fails.
```

> - AI Assistant may not preserve results of previously executed commands of the same type. Specify if the AI should preserve or discard previous results in your requests (include keywords such as "also" or "only").

Reviewer: ```suggestion
- The AI Assistant may not preserve the results of previously executed commands of the same type. To preserve or discard previous results, specify this in your request with keywords such as "also" or "only".
```

> - AI Assistant may not preserve results of previously executed commands of the same type. Specify if the AI should preserve or discard previous results in your requests (using keywords such as "also" or "only").

Reviewer 2: ```suggestion
- AI Assistant may not preserve results of previously executed commands of the same type. Specify if the AI should preserve or discard previous results in your requests (include keywords such as "also" or "only").
```

> - If {WidgetName} is bound to a large dataset, the `selectAll` command may increase the context size of requests beyond the limits of your AI service. The `selectAll` command adds all row keys to the request [context](/Documentation/ApiReference/UI_Components/dxDataGrid/Types/AIAssistantRequestCreatingEvent/#context) (in [onAIAssistantRequestCreating](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onAIAssistantRequestCreating)). To avoid this behavior, you can enable **selection**.[deferred](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/selection/#deferred).

Reviewer 2: ```suggestion
- If {WidgetName} is bound to a large dataset, the `selectAll` command may increase the context size of requests beyond the limits of your AI service. The `selectAll` command adds all row keys to the request [context](/Documentation/ApiReference/UI_Components/dxDataGrid/Types/AIAssistantRequestCreatingEvent/#context) (in [onAIAssistantRequestCreating](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onAIAssistantRequestCreating)). To avoid this behavior, you can activate **selection**.[deferred](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/selection/#deferred).
```

> - The assistant does not support certain actions that are only accessible in the component UI (for instance, expanding/collapsing groups).

Reviewer 2: ```suggestion
- The assistant does not support certain actions that are only accessible in the component UI (for instance, it cannot expand/collapse groups).
```

> - If you expect users to request certain commands in the AI Assistant, ensure the corresponding {WidgetName} feature is enabled to avoid command failures.

Reviewer: ```suggestion
- If users are likely to request specific commands in the AI Assistant, ensure that the corresponding {WidgetName} feature is enabled to prevent command failures.
```
